tests/test_autonomy/test_images/base.py

```diff
@@ -71,3 +80,11 @@ def build_image(
                     output += stream_data["status"]
 
         return True, output
+
+    def teardown(self) -> None:
+        """Teardown test."""
+
+        for container in self.running_containers:
+            with contextlib.suppress(APIError):
+                container.kill()
+                container.wait(timeout=30)
```

Comment:
Proper teardown of the docker images

Reply:
The test failure here https://github.com/valory-xyz/open-autonomy/actions/runs/4202934621/jobs/7294452747 was caused by the docker image not running when performing a clean up which might have caused by the agent crashing after the starting. We only care if the agent started and entered in the registration startup behaviour, we perform this check here https://github.com/valory-xyz/open-autonomy/blob/397431b265799339b5b24b6d050e9789d8d68d63/tests/test_autonomy/test_images/test_runtime.py#L127 so no need to worry if the agent crashed afterwards